### PR TITLE
trigger a full rebuild when the build script changes

### PR DESCRIPTION
**Pull request by jakemac53**
closes https://github.com/dart-lang/build/issues/53

### CODE REVIEW

File: lib/src/generate/build_impl.dart [hunk 1]
```diff
@@ -120,6 +133,51 @@ class BuildImpl {
     }
   }
 
+  /// Checks if the [_assetGraph] needs to be completely invalidated.
+  ///
+  /// For now this just means looking at the current running program, and seeing
+  /// if any of its sources are newer than the asset graph itself.
+  Future<bool> _shouldInvalidateAssetGraph() async {
+    var shouldRebuild = false;
+    await Future.wait(currentMirrorSystem().libraries.keys.map((Uri uri) async {
```
Comment (kegluneq) [inline lib/src/generate/build_impl.dart]: Once `shouldRebuild == true`, we should be able to return `true` without waiting for the results of other futures.

Let's discuss offline.

File: test/common/in_memory_reader.dart [hunk 2]
```diff
@@ -35,9 +36,29 @@ class InMemoryAssetReader implements AssetReader {
     }
   }
 
+  Map<Uri, LibraryMirror> __allLibraries;
```
Comment (kegluneq) [inline test/common/in_memory_reader.dart]: Why `__`?

Comment (jakemac53) [inline test/common/in_memory_reader.dart]: changed to just a final var

File: lib/src/generate/build_impl.dart (hunk 1 above)
Comment (jakemac53) [inline lib/src/generate/build_impl.dart]: Will switch to use a completer

**Comment by jakemac53** [comment]
updated

**Comment by kegluneq** [comment]
LGTM

**Pull request merged**